### ISSUE TITLE
Publish each component as an individual package

### ISSUE THREAD

**Issue by titouanmathis** [studiometa/ui#72]
It might become problematic eventually to publish every component in a single package, as a breaking change in one component could prevent upgrading to a newer version to fix another one.

- [ ] Identify the pros and cons

**Comment by KeziahMoselle** [comment]
I agree that separate packages for components would be useful for specific use cases/projects, maybe moving to a monorepo would be a great move. But as you said it involves several issues as well.

Maybe a better solution for maintainability would be to only publish individual components (like https://reach.tech/), then maybe we can have versioned documentation for each package on the main documentation and instructions on how to install it/use it.

**Comment by perruche** [comment]
**Cons**. 
- Splitting components in individual package seems like a lot of work to setup and to maintain. 
- Documentation will cause problems, which version of the component will be in the doc ?

**Questions**. 
- Is there a request from the team to move to this architecture ? 
- Have we encounter issues to upgrade ?
- Overriding specific component is already possible with current architecture and has been done several times with this pattern: `import { Slider as SliderCore } from '@studiometa/ui';` 
- Should we update the package more frequently on projects where we do maintenance ? As in update every project when releasing a new version.

**Comment by titouanmathis** [comment]
> - Is there a request from the team to move to this architecture ?

Not really, this is something I have been thinking about for a few months. I encountered some issues with side effects when updating in some projects. 

> - Overriding specific component is already possible with current architecture and has been done several times with this pattern: import { Slider as SliderCore } from '@studiometa/ui';

The issue is not about overriding specific components, but more about potentially introducing a breaking change in a component when updating the library to fix another one. 

I have `A` and `B` components in my library version 1.1.1, and I publish a new version including a breaking change for the component `A` and a patch for the component `B`. As this version contains a breaking change, it will be v2.0.0. 

Now in my project, I need to update the library because I need the patch on component `B`. This means that to fix a minor bug, I have to also update the project's implementation of the component `A` because of the breaking change. 

Potential solutions are:

- publish each component as their own package
- apply patches on multiple major versions
- be cautious when updating the package to apply a patch of its possible side effects on other components

> - Should we update the package more frequently on projects where we do maintenance ? As in update every project when releasing a new version.

This would be a best practice, I am not sure how we can manage it today.
